### PR TITLE
12720 list invoices project view

### PR DESCRIPTION
**Pull request by allthesignals**
# Big Picture Summary
Adds the UI for fetching the payment link for invoices. Lists invoices and links to their show view.

### Which major feature does this fit into?
Payment

### What was wrong and how does this fix the problem?
No way to pay — this adds the UI to interact with the payment endpoint.

## Technical Explanation — How does it work?
Just some components. The component hits the altready-built citypay service in the backend, which generates a citypay link. This link is sent to the user client.

Want to get some eyes on this, will fill out test coverage tomorrow.

Branches off some of @dhochbaum-dcp work.

### CODE REVIEW

File: client/app/components/project/package-section.hbs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      <ol class="no-bullet">
+        {{#each @packages as |package|}}
+          <Project::PackageListItem
+            @package={{package}}
+          />
+        {{/each}}
+      </ol>
+    </div>
+    <div class="cell large-6">
+      {{#let (flatten (map-by 'invoices' @packages)) as |invoices|}}
```
Comment (godfreyyeung) [inline client/app/components/project/package-section.hbs:15]: Hmm why flatten? After map by, does(n't) it look like this? `[{ dcpGrandTotal: 323, dcpName: 'name'}, {dcpGrandTotal: 342, dcpName: 'name2'....]`

Comment (godfreyyeung) [inline client/app/components/project/package-section.hbs:15]: Hmm. Each package can have multiple invoices, but each invoice is in an array itself?

File: client/app/components/project/package-section.hbs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+            {{#each invoices as |invoice|}}
+              <Project::PackageSection::InvoiceInfo
+                @invoice={{invoice}}
+              />
+            {{/each}}
+
+            <br/>
+            <br/>
+            <h3>Total: ${{reduce (fn this.sum) 0 (map-by 'dcpGrandtotal' invoices)}}</h3>
```
Comment (godfreyyeung) [inline client/app/components/project/package-section.hbs:49]: would this be more clear if it was all done in a getter/cp?

**Review by godfreyyeung — APPROVED** [review]

File: client/app/components/project/package-section.hbs (hunk 1 above)
Comment (godfreyyeung) [reply client/app/components/project/package-section.hbs:15]: Having a hard time visualizing the before/after of this data structure

Comment (allthesignals) [reply client/app/components/project/package-section.hbs:15]: no, that's not what this says. each package has many invoices. this is mapping out invoices from package _s_, plural.

File: client/app/components/project/package-section.hbs (hunk 2 above)
Comment (allthesignals) [reply client/app/components/project/package-section.hbs:49]: Just for the sake of discussion on handlebars helpers, can you try to make a guess at what this does?

File: client/app/components/project/package-section.hbs (hunk 1 above)
Comment (godfreyyeung) [reply client/app/components/project/package-section.hbs:15]: @allthesignals  Hmm, I was under the assumption we should list invoices by package. Just realized we have a bit of discrepancy between our current wireframe design and the wording on acceptance criteria. Maybe it's fine though

File: client/app/components/project/package-section.hbs (hunk 2 above)
Comment (godfreyyeung) [reply client/app/components/project/package-section.hbs:49]: It's basically a reduce function and you're supplying the accumulator function (simple add). It just looks a tiny bit wonky at first and could probably be done in a getter 

So it's adding up each dcpGrandtotal value on each invoice

File: client/app/components/project/package-section.hbs (hunk 1 above)
Comment (allthesignals) [reply client/app/components/project/package-section.hbs:15]: > list invoices by package

@godfreyyeung hmm, I'm not sure how what I said contradicts this? It does list invoices by package. As far as I understand, there can't be invoices for draft packages.

Comment (allthesignals) [reply client/app/components/project/package-section.hbs:15]: Let me know if we should huddle on this @godfreyyeung

**Pull request merged**